app/services/daily_usages/fill_history_service.rb

```diff
@@ -56,6 +56,10 @@ def call
         end
       end
 
+      if subscription.terminated?
```

Comment:
Same concern as above :)

Reply:
Here we are fine as the `to` value in the loop is `(subscription.terminated_at || Time.current).to_date` of if the subscription is terminated, we have to compute the value for the termination day